### PR TITLE
Animator refactor

### LINKED ISSUE #2269: Animators refactor
- [ ] The Animators all extend from Plots.Animator which seems really odd.
- [ ] The main Animator that a bunch of the Plots use is called Animators.Base, which is not a good name.

Changes: 
- `.iterativeDelay()` -> `.stepDelay()`
- `.easing()` -> `.easingMode()`
- `DEFAULT_EASING` -> `DEFAULT_EASING_MODE`
- `Animators.Plot` -> `Animator`
- `Animators.Base` -> `Animators.Easing`

### PR DESCRIPTION
**Pull request by aicioara**
Closes #2269 
Changes:
- `Animators.Base.iterativeDelay()` -> `Animators.Base.stepDelay()`
- `Animators.Base.easing()` -> `Animators.Base.easingMode()`
- `DEFAULT_START_DELAY_MILLISECONDS` -> DELETED (privatized)
- `DEFAULT_STEP_DURATION_MILLISECONDS` -> DELETED (privatized)
- `DEFAULT_ITERATIVE_DELAY_MILLISECONDS` -> DELETED (privatized)
- `DEFAULT_MAX_TOTAL_DURATION_MILLISECONDS` -> DELETED (privatized)
- `DEFAULT_EASING_MODE` -> DELETED (privatized)
- `Animators.Plot` -> `Animator`
- `Animators.Base` -> `Animators.Easing`
- Shifting files around 

**TODO:**
- `base_animator.js` -> `easing_animator.js` Did not do it yet because it is easier to check overlaying tests

**QE Notes:** Regression pass. Everything should work as before

### CODE REVIEW

**Comment by bluong** [comment]
Does `Animators.Easing` make sense?  It doesn't make sense to me at all to me at first glance :/

**Comment by aicioara** [comment]
Yes, we talked over this for quite a bit. Alternatives are:
- `Animators.General` 
- `Animators.Default`
- `Animators.Base` - old version
- `Animator` - Conflict with the interface
- `Animators.Easing` - top Choice

The reason why easing makes sense is because that is the type of animation this is implementing. It is also in symmetry with `Animators.Null`, which is a no-op. the `Null` bit represents the no-op as an animation type, so `Easing` also represents the animation type here

**Comment by bluong** [comment]
Consider writing an RFC for the naming of the class.  Your logic seems sound, but it still strikes me as a bit odd.

File: quicktests/overlaying/tests/functional/base_animator.js [hunk 1]
```diff
@@ -17,10 +17,10 @@ function run(svg, data, Plottable) {
 
     var xAxis = new Plottable.Axes.Category(xScale, "bottom");
     var yAxis = new Plottable.Axes.Numeric(yScale, "left");
-    var animator = new Plottable.Animators.Base();
+    var animator = new Plottable.Animators.Easing();
         animator.stepDuration(1000);
         animator.maxTotalDuration(2000);
-        animator.iterativeDelay(100);
+        animator.stepDelay(100);
```
Comment (bluong) [inline quicktests/overlaying/tests/functional/base_animator.js]: Gonna need to writing an API-breaking guard around this

File: src/animators/easingAnimator.ts [hunk 2]
```diff
@@ -6,45 +6,45 @@ export module Animators {
...
     /**
      * The default maximum total animation duration
      */
-    public static DEFAULT_MAX_TOTAL_DURATION_MILLISECONDS = Infinity;
+    private static DEFAULT_MAX_TOTAL_DURATION_MILLISECONDS = Infinity;
     /**
      * The default easing of the animation
      */
-    public static DEFAULT_EASING = "exp-out";
+    private static DEFAULT_EASING_MODE = "exp-out";
```
Comment (bluong) [inline src/animators/easingAnimator.ts]: underscores please

**Comment by aicioara** [comment]
BLUF: We are considering changing the Animators.Base to something more appropriate.

We are considering renaming the Base Animator Animators.Base to an Easing Animator Animators.Easing. 

There are multiple reasons for that. 
There is nothing that extends the Base Animator at the moment.
`Easing` makes sense is because that is the type of animation this is implementing. http://easings.net/
It is in symmetry with `Animators.Null`, which is a no-op. the `Null` bit represents the no-op as an animation type, so `Easing` also represents the animation type here
However, it is a bit weird that the Easing animator also has logic for the steps timing and inter step delays.

Alternatives we have considered are
`Animators.General` 
`Animators.Default`
`Animators.Base` - old version
`Animator` - Conflict with the interface
`Animators.Easing` - top Choice
`Animators.Step`
`Animators.Iterative`

Unless there are better alternatives or strong opposition, we plan to stick with `Animators.Easing`.

Cheers,
Andrei @ Plottable

**Pull request merged**